### PR TITLE
Support OpenFin Platform

### PR DESCRIPTION
**Pull request by ilzep**
When using OpenFin platform, it is not recommended to use the platform provider window, instead define a child window. Update the logic of loading layout, to make sure this window is not closed as one of the child windows. 

Added sample platform config in examples.

### CODE REVIEW

File: examples/openfin/platform.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                "defaultHeight": 500,
+                "saveWindowState": false,
+                "defaultCentered": true,
+                "url": "http://localhost:8000/index.html"
+            }
+        ]
+    },
+    "runtime": {
+        "arguments": "",
+        "version": "22.94.65.4"
```
Comment (bingenito) [inline examples/openfin/platform.json:23]: should we just point at stable for the example?

File: examples/openfin/platform.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (bingenito) [inline examples/openfin/platform.json:1]: do we still need app.json?

Comment (bingenito) [reply examples/openfin/platform.json:1]: Actually it is nice being able to test both classic and platform

File: examples/openfin/platform.json (hunk 1 above)
Comment (ilzep) [reply examples/openfin/platform.json]: I have update to target stable instead. This was left over from testing

File: examples/openfin/platform.json (hunk 2 above)
Comment (ilzep) [reply examples/openfin/platform.json:1]: Agreed, I think we want to be able to test both configs.

**Review by bingenito — APPROVED** [review]

**Review by psmulovics — APPROVED** [review]

**Pull request merged**